### DISCUSSION TITLE
Move the controls in [Youtube.com] from the bottom of the page to the top of the page.

### DISCUSSION THREAD

**Discussion by Docfxit** [openstyles/stylus#1835] [Q&A]
I would like to move the controls in [Youtube.com] from the bottom of the page to the top of the page.
A lot of people creating Youtube videos put text on the bottom of the page and the controls cover it up when I pause a video. 
I created code to move the controls to the top but it doesn't work:
`/* Apply styles when the "yt-controls-top" class is present on the body */
[http://body.yt](http://disq.us/url?url=http%3A%2F%2Fbody.yt%3A1-qFps0ea4pSS6EEVV2A9ec8tug&cuid=3760007) -controls-top .ytp-chrome-bottom {
position: absolute !important;
top: 10px !important;
bottom: auto !important;
width: 100% !important;
z-index: 1000 !important;
}

[http://body.yt](http://disq.us/url?url=http%3A%2F%2Fbody.yt%3A1-qFps0ea4pSS6EEVV2A9ec8tug&cuid=3760007) -controls-top .ytp-gradient-bottom {
display: none !important;
}

[http://body.yt](http://disq.us/url?url=http%3A%2F%2Fbody.yt%3A1-qFps0ea4pSS6EEVV2A9ec8tug&cuid=3760007) -controls-top .ytp-chrome-controls {
align-items: flex-start !important;
}`
I selected URL and put in https://www.youtube.com/
I'm using Firefox ver: 132.0.1
Stylus is up-to-date
Can anyone figure out what is wrong with the code?

**Comment by tophf** [comment]
This is invalid syntax. Why is there an http URL and what is `-controls-top`?

Something like this may suffice:
```css
@-moz-document url-prefix("https://www.youtube.com/watch") {
.ytp-chrome-bottom {
  bottom: auto;
}
}
```

**Reply by Docfxit** [reply]
Thank you for the suggestion.  
It doesn't seem to do anything.  
What I'm trying to accomplish is to move all the controls from the bottom of the screen to the top of the screen.
I'm new to Stylus so maybe I'm not doing something I should.
After installing the code I refreshed the Youtube screen.
I turned Stylus plugin off and then on.
I closed Firefox and re-opened it.
 
![YoutubeCode](https://github.com/user-attachments/assets/7c42d5dd-2895-41f7-99e0-e38296323c51)
![YoutubeInstalled](https://github.com/user-attachments/assets/afbf9493-4a25-4a43-9f00-3d64be4e447a)

**Reply by tophf** [reply]
It works for me, so maybe our understanding of what "controls" are is different...

**Reply by Docfxit** [reply]
Thank you very much for looking into this:
![YoutubeControls](https://github.com/user-attachments/assets/0045cfb0-af39-4e67-a88f-fdf89c092a32)
That's the location on the screen where a lot of Youtube videos put comments.  If I could move those to the top, I'd be able to read the comments when I pause the video.

**Reply by tophf** [reply]
Okay, that's what I thought too, and my style moves them to the top. Maybe you have another userscript in Tampermonkey?

**Reply by Docfxit** [reply]
I'm not using Tampermonkey.
Are you using Firefox?
Are you running Windows 10?

**Reply by tophf** [reply]
Yeah and yeah. It's the same in Chrome and Firefox. Maybe Youtube is serving a different version of their site, they do that regularly on random.

![image](https://github.com/user-attachments/assets/d8379b1e-d00f-487e-99e8-d78de536129c)

**Reply by tophf** [reply]
See if you have granted site permissions to the extension somewhere in its icon's right-click menu or in `about:addons`.

**Reply by Docfxit** [reply]
![StylusPermissions](https://github.com/user-attachments/assets/ffcddf67-da4f-4e05-b72d-b9a7e80971bd)
I don't know if there is anything I can show you?
![YoutubeInspect](https://github.com/user-attachments/assets/e46c44ed-62c3-4cff-919f-3fdccee8c8b1)

**Reply by tophf** [reply]
Everything looks fine to me. Try replacing `auto` with `auto !important`

**Reply by Docfxit** [reply]
That's super.  That fixed it.  You are Great.  Thank you very very much.

I have a bookmark that is supposed to toggle the controls from the top to bottom.  It's not working.
Bookmark name: Toggle YouTube Controls
javascript:(function(){      document.body.classList.toggle('yt-controls-top');  })();

Do you have any idea why?

**Reply by tophf** [reply]
Replace `.ytp-chrome-bottom` with `[yt-controls-top] .ytp-chrome-bottom`

**Reply by Docfxit** [reply]
It worked the first time I did it.  After that it won't work.  I think I did something wrong.
This is what I have now:
javascript:(function(){      document.body.classList.toggle('.ytp-chrome-bottom');  })();

**Reply by tophf** [reply]
I made a mistake. Revert your code (remove the dot) and use `.yt-controls-top .ytp-chrome-bottom` in the style.

**Reply by Docfxit** [reply]
It's not working.  I probably made a mistake.
This is what I have for the bookmark:
javascript:(function(){      document.body.classList.toggle('.yt-controls-top .ytp-chrome-bottom');  })();

**Reply by tophf** [reply]
No, use your original code for the bookmark: `javascript:(function(){ document.body.classList.toggle('yt-controls-top'); })();`
And `.yt-controls-top .ytp-chrome-bottom {` in the style

**Reply by Docfxit** [reply]
Now that's magical.  It works great now.
Can I buy you a drink?
You are super.....
